drivers/dma/dma_mcux_lpc.c

```diff
@@ -281,8 +281,14 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 	dev_config = dev->config;
 	dma_data = dev->data;
 	block_config = config->head_block;
-	width = MIN(config->source_data_size, config->dest_data_size);
-	max_xfer = NXP_LPC_DMA_MAX_XFER * width;
+	/* The DMA controller deals with just one transfer
+	 * size, though the API provides separate sizes
+	 * for source and dest. So assert that the source
+	 * and dest sizes are the same.
+	 */
+	assert(config->dest_data_size == config->source_data_size);
```

Comment:
This assert may cause failures in some of the other drivers that use this driver. I would suggest removing it and updating the comment.

Reply:
If the caller has the dest_data_size == source_data_size, then the assert won't trip, and it won't trip if asserts are not enabled.

If the caller has different values for src and dest sizes, then it's a problem they need to think about and fix.
The old driver (prior to the change to support chaining for long transfers) used MIN. The driver
that supports the long transfers used MAX. This changing of behavior already broke existing drivers silently (e.g. the I2S usage). I think it'd be better to detect such problems as soon as possible, instead of silently doing something they might not be expecting and possibly setting the size the opposite of what the caller intended.

Reply:
I know we will have to update the SPI driver to address this. Let me know if you would like me to send you a patch.

Reply:
I can add that. I thought I had checked the SPI driver, since we use that too, but I think we didn't catch it because our usage of SPI is always using 1-byte dest_data_size (src_data_size is fixed at 1).

Reply:
I think also the UART driver will be fine because destination and source data sizes will both be 1 always